OpenSim/Common/Path.h

```diff
@@ -0,0 +1,152 @@
+#ifndef OPENSIM_PATH_H_
+#define OPENSIM_PATH_H_
+/* -------------------------------------------------------------------------- *
+ *                           OpenSim: Path.h                                  *
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2016 Stanford University and the Authors                *
+ * Author(s): Carmichael Ong                                                  *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "OpenSim/Common/Object.h"
+
+namespace OpenSim {
+
+//==============================================================================
+//                                 OPENSIM PATH
+//==============================================================================
+/**
+* An abstract class for handling a Path. A Path refers to a list of strings that
+* represent a different level in a hierarchical structure. Each level is divided
+* by designated separators (e.g., "/" or "\").
+*
+* This class stores the list of levels as a vector of strings. One char is used
+* to denote a path separator when either reading in or writing the Path out to
+* a string. A bool is stored to determine whether this Path should be resolved
+* relative to the root (denoted with a leading separator char) or not.
+*
+* @author Carmichael Ong
+*/
+class OSIMCOMMON_API Path : public Object {
+    OpenSim_DECLARE_ABSTRACT_OBJECT(Path, Object);
```

Comment:
My preference is to have `Path` be a built in type rather than an Object. It would be nice to have a corresponding `Property<Path>` as a SimpleProperty vs an ObjectProperty for serialization.